### PR TITLE
DOC: Improved docs for `jnp.fft.fft` and `jnp.fft.ifft`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
This PR includes modified docstrings for `jnp.fft.fft` and `jnp.fft.ifft`. Also changes the description of the APIs `jnp.fft.fftn` and `jnp.fft.ifftn`.

Part of #21461

### CODE REVIEW

File: jax/_src/numpy/fft.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def _fft_core(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
 def fftn(a: ArrayLike, s: Shape | None = None,
          axes: Sequence[int] | None = None,
          norm: str | None = None) -> Array:
-  r"""Compute discrete Fourier transform of multidimensional array along given axes.
+  r"""Compute multidimensional discrete Fourier transform along given axes.
```
Comment (dfm) [inline jax/_src/numpy/fft.py:115]:
```suggestion
  r"""Compute a multidimensional discrete Fourier transform along given axes.
```

File: jax/_src/numpy/fft.py [hunk 2]
```diff
@@ -186,7 +186,7 @@ def fftn(a: ArrayLike, s: Shape | None = None,
 def ifftn(a: ArrayLike, s: Shape | None = None,
           axes: Sequence[int] | None = None,
           norm: str | None = None) -> Array:
-  r"""Compute inverse discrete Fourier transform of multidimensional array.
+  r"""Compute multidimensional inverse discrete Fourier transform along given axes.
```
Comment (dfm) [inline jax/_src/numpy/fft.py:189]:
```suggestion
  r"""Compute a multidimensional inverse discrete Fourier transform along given axes.
```

File: jax/_src/numpy/fft.py [hunk 3]
```diff
@@ -126,15 +126,15 @@ def fftn(a: ArrayLike, s: Shape | None = None,
...
+    An array containing the multidimensional discrete Fourier transform of ``a``.
 
   See also:
-    - :func:`jax.numpy.fft.fft`: Computes the discrete Fourier transform of
-      one-dimensional array.
-    - :func:`jax.numpy.fft.ifft`: Computes the inverse of discrete Fourier transform
-      of one-dimensional array.
-    - :func:`jax.numpy.fft.ifftn`: Computes the inverse of discrete Fourier transform
-      of multidimensional array.
+    - :func:`jax.numpy.fft.fft`: Computes one-dimensional discrete Fourier
```
Comment (dfm) [inline jax/_src/numpy/fft.py:132]:
```suggestion
    - :func:`jax.numpy.fft.fft`: Computes a one-dimensional discrete Fourier
```

And the ones below... 👇

File: jax/_src/numpy/fft.py [hunk 4]
```diff
@@ -200,15 +200,16 @@ def ifftn(a: ArrayLike, s: Shape | None = None,
...
+    of ``a``.
 
   See also:
-    - :func:`jax.numpy.fft.fftn`: Computes the discrete Fourier transform of
-      multidimensional array.
-    - :func:`jax.numpy.fft.fft`: Computes the discrete Fourier transform of
-      one-dimensional array.
-    - :func:`jax.numpy.fft.ifft`: Computes the inverse of discrete Fourier transform
-      of one-dimensional array.
+    - :func:`jax.numpy.fft.fftn`: Computes multidimensional discrete Fourier
```
Comment (dfm) [inline jax/_src/numpy/fft.py:207]:
```suggestion
    - :func:`jax.numpy.fft.fftn`: Computes a multidimensional discrete Fourier
```

👇

File: jax/_src/numpy/fft.py [hunk 5]
```diff
@@ -281,15 +282,120 @@ def _fft_core_1d(func_name: str, fft_type: xla_client.FftType,
   return _fft_core(func_name, fft_type, a, s, axes, norm)
 
 
-@implements(np.fft.fft)
 def fft(a: ArrayLike, n: int | None = None,
         axis: int = -1, norm: str | None = None) -> Array:
+  r"""Compute one-dimensional discrete Fourier transform along a given axis.
```
Comment (dfm) [inline jax/_src/numpy/fft.py:287]:
```suggestion
  r"""Compute a one-dimensional discrete Fourier transform along a given axis.
```

File: jax/_src/numpy/fft.py [hunk 6]
```diff
@@ -281,15 +282,120 @@ def _fft_core_1d(func_name: str, fft_type: xla_client.FftType,
...
+    Array(True, dtype=bool)
+  """
   return _fft_core_1d('fft', xla_client.FftType.FFT, a, n=n, axis=axis,
                       norm=norm)
 
-@implements(np.fft.ifft)
+
 def ifft(a: ArrayLike, n: int | None = None,
          axis: int = -1, norm: str | None = None) -> Array:
+  r"""Compute one-dimensional inverse discrete Fourier transform along given axis.
```
Comment (dfm) [inline jax/_src/numpy/fft.py:350]:
```suggestion
  r"""Compute a one-dimensional inverse discrete Fourier transform along given axis.
```

File: jax/_src/numpy/fft.py [hunk 7]
```diff
@@ -281,15 +282,120 @@ def _fft_core_1d(func_name: str, fft_type: xla_client.FftType,
...
+    axis: int, default=-1. Specifies the axis along which the transform is computed.
+      If not specified, the transform is computed along axis -1.
+    norm: string. The normalization mode. "backward", "ortho" and "forward" are
+      supported.
+
+  Returns:
+    An array containing the one-dimensional discrete Fourier transform of ``a``.
+
+  See also:
+    - :func:`jax.numpy.fft.ifft`: Computes one-dimensional inverse discrete
```
Comment (dfm) [inline jax/_src/numpy/fft.py:304]:
```suggestion
    - :func:`jax.numpy.fft.ifft`: Computes a one-dimensional inverse discrete
```

👇

File: jax/_src/numpy/fft.py [hunk 8]
```diff
@@ -281,15 +282,120 @@ def _fft_core_1d(func_name: str, fft_type: xla_client.FftType,
...
+    axis: int, default=-1. Specifies the axis along which the transform is computed.
+      If not specified, the transform is computed along axis -1.
+    norm: string. The normalization mode. "backward", "ortho" and "forward" are
+      supported.
+
+  Returns:
+    An array containing the one-dimensional discrete Fourier transform of ``a``.
+
+  See also:
+    - :func:`jax.numpy.fft.fft`: Computes one-dimensional discrete Fourier transform.
```
Comment (dfm) [inline jax/_src/numpy/fft.py:367]:
```suggestion
    - :func:`jax.numpy.fft.fft`: Computes a one-dimensional discrete Fourier transform.
```

👇

**Review by dfm — CHANGES_REQUESTED** [review]
Thanks! I added some small comments. Basically, I think that we always want to say "a one-dimensional Fourier transform" or "the one-dimensional Fourier transform". I don't feel very strongly about which one to prefer ("a" vs. "the").

**Comment by rajasekharporeddy** [comment]
Modified the docstrings as suggested and squashed all commits into one.

**Review by dfm — APPROVED** [review]

File: jax/_src/numpy/fft.py [hunk 9]
```diff
@@ -200,15 +200,16 @@ def ifftn(a: ArrayLike, s: Shape | None = None,
...
-    - :func:`jax.numpy.fft.fft`: Computes the discrete Fourier transform of
-      one-dimensional array.
-    - :func:`jax.numpy.fft.ifft`: Computes the inverse of discrete Fourier transform
-      of one-dimensional array.
+    - :func:`jax.numpy.fft.fftn`: Computes multidimensional discrete Fourier
+      transform.
+    - :func:`jax.numpy.fft.fft`: Computes one-dimensional discrete Fourier
+      transform.
+    - :func:`jax.numpy.fft.ifft`: Computes one-dimensional inverse discrete
+      Fourier transform.
```
Comment (dfm) [inline jax/_src/numpy/fft.py:212]: These lines should also all say "a" or "the".

**Review by dfm — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/fft.py (hunk 9 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Done

**Comment by rajasekharporeddy** [comment]
Rebased onto the main.

**Pull request merged**